app/models/backend/transaction.rb

```diff
@@ -1,102 +1,43 @@
+# frozen_string_literal: true
+
+# Used by Stats tests and backend random data generator
 module Backend
-  class Transaction # this stuff is used by the stats tests, beware
-    attr_accessor :estimated_usage
-    attr_accessor :id
-    attr_accessor :provider_account_id
-    attr_accessor :created_at
-    #TODO: there's some duplication between this and delegate :to => :cinstance_data
-    attr_accessor :service_id
-    attr_accessor :client_ip
-    attr_accessor :cinstance_id
-    attr_accessor :log          # hash with the log info
+  class Transaction
+    attr_accessor :id, :estimated_usage, :provider_account_id, :created_at, :service_id, :client_ip, :cinstance_id, :log
 
     attr_writer :user_key
-    attr_writer :confirmed
 
-    delegate :provider_verification_key, :plan_name, :application_id,
-             :service_id, :to => :cinstance_data
+    delegate :provider_verification_key, :plan_name, :application_id, to: :cinstance_data
 
     def initialize(attributes = {})
-      @confirmed = false
-
       self.attributes = attributes
     end
 
-    def self.find(service_id, id)
-      find_by_id(service_id, id) || raise(TransactionNotFound)
-    end
-
-    def self.find_by_id(service_id, id)
-      attributes = storage.get_object(storage_key(service_id, id))
-      attributes && new(attributes.merge(:id => id))
-    end
-
     def self.report!(params = {})
       transaction = new(params)
       transaction.report!
       transaction
     end
 
-    # Report multiple transactions at the same time.
-    #
-    # Transaction.report_multiple!(@provider_account.id,
-    #                              '0' => {:user_key => @user_key_one,
-    #                                      :usage => {'hits' => 1},
-    #                                      :client_ip => "1.2.3.4"},
-    #                              '1' => {:user_key => @user_key_two,
-    #                                      :usage => {'hits' => 1},
-    #                                      :client_ip => "1.2.3.5"})
-    def self.report_multiple!(provider_account_id, raw_transactions)
-      reporter = MultiReporter.new(provider_account_id)
```

Comment:
`MultiReporter` class is missing, and this method was not used anywhere.

Reply:
Searching for a different thing, I found MultiReporter :sweat_smile: https://github.com/3scale/system/commit/dc3d69a64c6d34020a3c6f1b8b406e44d013fff1#diff-0e21d54a70c962b4129df4dc752c7380105575cdc93272d30502b0e394a40e5c